commons/deployment/src/main/java/com/github/mcollovati/quarkus/hilla/deployment/QuarkusHillaExtensionProcessor.java

```diff
@@ -284,12 +284,26 @@ void registerEndpoints(
     }
 
     @BuildStep
-    void registerHillaPushServlet(BuildProducer<ServletBuildItem> servletProducer, NativeConfig nativeConfig) {
+    void registerHillaPushServlet(
+            BuildProducer<ServletBuildItem> servletProducer,
+            QuarkusEndpointConfiguration endpointConfiguration,
+            NativeConfig nativeConfig) {
         ServletBuildItem.Builder builder = ServletBuildItem.builder(
                 QuarkusAtmosphereServlet.class.getName(), QuarkusAtmosphereServlet.class.getName());
-        builder.addMapping("/HILLA/push")
+        String prefix = endpointConfiguration.getEndpointPrefix();
+        if (prefix.matches("^/?connect/?$")) {
+            prefix = "/";
+        } else if (!prefix.startsWith("/")) {
+            prefix = "/" + prefix;
+        }
+        if (prefix.endsWith("/")) {
+            prefix = prefix.substring(0, prefix.length() - 1);
```

Comment:
Is the case relevant, when multiple `/`are present at the beginning or ending?

Reply:
Many / is potentially a configuration error. We can throw an exception

Reply:
I suggest not complicating things and just handling leading and trailing slash.

Reply:
I think thats fine. As you mentioned would be configuration error.